Update legal.md (#3086)
github/opensource.guide: _articles/legal.md

## Understanding the legal implications of open source

> Sharing your creative work with the world can be an exciting and rewarding experience. It can also mean a bunch of legal things you didn't know you had to worry about. Thankfully, you don't have to start from scratch. We've got your legal needs covered. (Before you dig in, be sure to read our [disclaimer](/notices/).)

Author: This article does not and cannot "cover the reader's legal needs". Rather it is a broad introduction to the legal side of OSS.

> Sharing your creative work with the world can be an exciting and rewarding experience. It can also mean a bunch of legal things you didn't know you had to worry about. Thankfully, with this legal crash course you don't have to start from scratch. (Before you dig in, be sure to read our [disclaimer](/notices/).)

Reviewer: I wouldn't call this a legal course, here's another idea:

```suggestion
Sharing your creative work with the world can be an exciting and rewarding experience. It can also mean a bunch of legal things you didn't know you had to worry about. Thankfully, with this guide you don't have to start from scratch. (Before you dig in, be sure to read our [disclaimer](/notices/).)
```

## Why do people care so much about the legal side of open source?

Glad you asked! When you make a creative work (such as writing, graphics, or code), that work is under exclusive copyright by default. That is, the law assumes that as the author of your work, you have a say in what others can do with it.

In general, that means nobody else can use, copy, distribute, or modify your work without being at risk of take-downs, shake-downs, or litigation.

> Open source is an unusual circumstance, however, because the author expects that others will use, modify, and share the work. But because the legal default is still exclusive copyright, you need to explicitly state these permissions in a document called a license.

Reviewer: I don't love this because it sounds like creating a document is what the user needs to do, which is definitely not the case. Here's an alternative:
```suggestion
Open source is an unusual circumstance, however, because the author expects that others will use, modify, and share the work. But because the legal default is still exclusive copyright, you need to explicitly give these permissions with a license.
```

Open source is an unusual circumstance, however, because the author expects that others will use, modify, and share the work. But because the legal default is still exclusive copyright, you need a license that explicitly states these permissions.
Open source is an unusual circumstance, however, because the author expects that others will use, modify, and share the work. But because the legal default is still exclusive copyright, you need to explicitly state these permissions in a document called a license.

> If you don't apply an open source license, everybody who contributes to your project also becomes an exclusive copyright holder of their work. That means nobody can use, copy, distribute, or modify their contributions -- and that "nobody" includes you.

Author: I felt like this paragraph came across as "too legalese". I tried to improve the flow from the previous paragraph and simplify the language without removing any critical ideas.

You're in luck, because today, open source licenses are standardized and easy to use. You can copy-paste an existing license directly into your project.

> [MIT](https://choosealicense.com/licenses/mit/), [Apache 2.0](https://choosealicense.com/licenses/apache-2.0/), and [GPLv3](https://choosealicense.com/licenses/gpl-3.0/) are the most popular open source licenses, but there are other options to choose from. You can find the full text of these licenses, and instructions on how to use them, on [choosealicense.com](https://choosealicense.com/).

Author: Added a source for license popularities (libraries.io by Tidelift).

> [MIT](https://choosealicense.com/licenses/mit/), [Apache 2.0](https://choosealicense.com/licenses/apache-2.0/), and [GPLv3](https://choosealicense.com/licenses/gpl-3.0/) are the [most popular open source licenses](https://libraries.io/licenses), but there are other options to choose from. You can find the full text of these licenses, and instructions on how to use them, on [choosealicense.com](https://choosealicense.com/).

Reviewer: libraries.io's stats (which are probably dominated by npm) don't support the statement! Subbing in a new resource that does, and remove "the most" to make it more likely to be true forever.
```suggestion
[MIT](https://choosealicense.com/licenses/mit/), [Apache 2.0](https://choosealicense.com/licenses/apache-2.0/), and [GPLv3](https://choosealicense.com/licenses/gpl-3.0/) are [popular open source licenses](https://innovationgraph.github.com/global-metrics/licenses), but there are other options to choose from. You can find the full text of these licenses, and instructions on how to use them, on [choosealicense.com](https://choosealicense.com/).
```

## Which open source license is appropriate for my project?

> If you're starting from a blank slate, it's hard to go wrong with the [MIT License](https://choosealicense.com/licenses/mit/). It's short, very easy to understand, and allows anyone to do anything so long as they keep a copy of the license, including your copyright notice. You'll be able to release the project under a different license if you ever need to.

Author: As I understand it, the goal of this first paragraph is to *quickly* give readers the name of license they can use and move on with their lives if they are tired of thinking about legal stuff.

This rephrasing puts the name of MIT in the first line instead of the second.

* **"Limited copyleft" licenses** (sometimes called "weak copyleft") are similar to permissive licenses -- generally your project can use whatever license you want -- but modifications to the library itself must be shared under an identical/compatible license. Common limited copyleft licenses include the [MPL 2.0](https://choosealicense.com/licenses/mpl-2.0/) and the [LGPL](https://choosealicense.com/licenses/lgpl-3.0/).
* **"Strong copyleft" licenses** also give you permission to use, modify, and share the library, but only if your project uses an identical or [compatible license](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses). Common strong copyleft licenses include [GPLv2](https://choosealicense.com/licenses/gpl-2.0), [GPLv3](https://choosealicense.com/licenses/gpl-3.0), and [AGPLv3](https://choosealicense.com/licenses/agpl-3.0).

> 

Author: The long parentheticals defining each class of license break the reader's concentration. By re-writing in bullet point form, there's more space to explain without creating a wall of text, *and* it matches the layout of the following *communities* section.

I also added details about "limited copyleft" licenses to bring greater awareness to their unique properties. I deliberately chose the term "limited copyleft" to emphasize the limited reach of their copyleft provisions (the distinguishing characteristic of these licenses). The more common term is "weak copyleft", but I have seen that term prompt faulty impressions that such licenses somehow only provide "weak/less-enforceable" copyright protections.

Reviewer: Get the parens aren't ideal but feel like the bullets make it less obvious what the connection between dependencies' licenses and your project are. And I don't think it's really necessary to introduce the concept of limited or weak copyleft at this point.

Author: Good feedback. My edits shift the focus back towards the ramifications of a dependency's license on the larger project.

I still think it's helpful to briefly distinguish between strong/limited copyleft licenses since their requirements for downstream project licensing differ so widely. That was a big point of confusion for me when I started out in open source. Additionally, including the distinction here prepares the reader to understand the discussion a few paragraphs later of why some companies want a "strong" copyleft license.

The linked article about the "additional rules" of limited copyleft licenses is from FOSSA, a blog about open-source licensing. I am not affiliated with them, just a fan of their detail and approachability.

* **"Strong copyleft" licenses** also give you permission to use, modify, and share the library, but only if your project uses an identical or [compatible license](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses). Common strong copyleft licenses include [GPLv2](https://choosealicense.com/licenses/gpl-2.0), [GPLv3](https://choosealicense.com/licenses/gpl-3.0), and [AGPLv3](https://choosealicense.com/licenses/agpl-3.0).

You may also want to consider the **communities** you hope will use and contribute to your project:

* **Do you want your project to be used as a dependency by other projects?** Probably best to use the most popular license in your relevant community. For example, [MIT](https://choosealicense.com/licenses/mit/) is the most popular license for [npm libraries](https://libraries.io/search?platforms=NPM).

> * **Do you want your project to appeal to large businesses?** A large business will likely want an express patent license from all contributors. In this case, [Apache 2.0](https://choosealicense.com/licenses/apache-2.0/) has you (and them) covered.

Author: Adding a use case for choosing a "limited copyleft" license.

> * **Do you want your project to appeal to large businesses?** A large business will likely want an express patent license from all contributors and the ability to use the project in proprietary software. In this case, the permissive [Apache 2.0](https://choosealicense.com/licenses/apache-2.0/) or the limited copyleft [MPL 2.0](https://choosealicense.com/licenses/mpl-2.0/) has you (and them) covered.

Reviewer: I don't like this change. The ability to use in proprietary software is also true of MIT, and is only one of the reasons permissive is preferred. I again also don't see any reason to introduce limited copyleft here.

I'd also tone down the original recommendation a bit; the years since this was written have shown even more that MIT's implied patent license is very rarely if ever a blocker. Here's a suggestion:

```suggestion
* **Do you want your project to appeal to large businesses?** A large business may be comforted by an express patent license from all contributors. In this case, [Apache 2.0](https://choosealicense.com/licenses/apache-2.0/) has you (and them) covered.
```

* **Do you want your project to appeal to large businesses?** A large business will likely want an express patent license from all contributors and the ability to use the project in proprietary software. In this case, the permissive [Apache 2.0](https://choosealicense.com/licenses/apache-2.0/) or the limited copyleft [MPL 2.0](https://choosealicense.com/licenses/mpl-2.0/) has you (and them) covered.
* **Do you want your project to appeal to contributors who do not want their contributions to be used in closed source software?** [GPLv3](https://choosealicense.com/licenses/gpl-3.0/) or (if they also do not wish to contribute to closed source services) [AGPLv3](https://choosealicense.com/licenses/agpl-3.0/) will go over well.

> Your **company** may have specific licensing requirements for its open source projects. For example, it may require a permissive license so that the company can use your project in the company's closed source product. Or your company may require a strong copyleft license and an additional contributor agreement (see below) so that only your company, and nobody else, can use your project in closed source software. Or your company may have certain needs related to standards, social responsibility, or transparency, any of which could require a particular licensing strategy. Talk to your [company's legal department](#what-does-my-companys-legal-team-need-to-know).

Author: Emphasizing that the terms of "limited copyleft" licenses, particularly the MPL 2.0, are compatible with usage in proprietary software (a common misunderstanding).

Reviewer: Missed this in my initial review. I'd also skip this. Again, limited copyleft isn't a concept that needs to be in this introductory material, and the edit isn't true except for file-based copyleft like MPL-2.0 (from experience, LGPL always requires closer scrutiny), and file-based copyleft is really not a concept that needs to be delved into here.

Author: Fair enough. Upon re-reading I'm also recognizing that this part of the article is best if it recommends a single concrete license for each situation, and the Apache-2.0 is certainly the industry's preferred choice for explicit patent grants.

**Your project's existing copyright holders.** If you're the sole contributor to your project then either you or your company is the project's sole copyright holder. You can add or change to whatever license you or your company wants to. Otherwise there may be other copyright holders that you need agreement from in order to change licenses. Who are they? [People who have commits in your project](https://github.com/thehale/git-authorship) is a good place to start. But in some cases copyright will be held by those people's employers. In some cases people will have only made minimal contributions, but there's no hard and fast rule that contributions under some number of lines of code are not subject to copyright. What to do? It depends. For a relatively small and young project, it may be feasible to get all existing contributors to agree to a license change in an issue or pull request. For large and long-lived projects, you may have to seek out many contributors and even their heirs. Mozilla took years (2001-2006) to relicense Firefox, Thunderbird, and related software.

> Alternatively, you can have contributors agree in advance (via an additional contributor agreement -- [see below](#does-my-project-need-an-additional-contributor-agreement)) to certain license changes under certain conditions, beyond those allowed by your existing open source license. This shifts the complexity of changing licenses a bit. You'll need more help from your lawyers up front, and you'll still want to clearly communicate with your project's stakeholders when executing a license change.

Reviewer 2: I suggest putting "see below" in brackets here too as you did below.

```suggestion
Alternatively, you can have contributors agree in advance via an additional contributor agreement ([see below](#does-my-project-need-an-additional-contributor-agreement)) to certain license changes under certain conditions, beyond those allowed by your existing open source license. This shifts the complexity of changing licenses a bit. You'll need more help from your lawyers up front, and you'll still want to clearly communicate with your project's stakeholders when executing a license change.
```

Author: Originally, I only linked up the existing text. In the latest edits, I took your re-write a few steps further to remove redundant clauses.

e.g. "under certain conditions" and "beyond those allowed ..." are already implied by the need for a separate contract.

> **Your project's existing copyright holders.** If you're the sole contributor to your project then either you or your company is the project's sole copyright holder. You can add or change to whatever license you or your company wants to. Otherwise there may be other copyright holders that you need agreement from in order to change licenses. Who are they? People who have commits in your project is a good place to start. But in some cases copyright will be held by those people's employers. In some cases people will have only made minimal contributions, but there's no hard and fast rule that contributions under some number of lines of code are not subject to copyright. What to do? It depends. For a relatively small and young project, it may be feasible to get all existing contributors to agree to a license change in an issue or pull request. For large and long-lived projects, you may have to seek out many contributors and even their heirs. Mozilla took years (2001-2006) to relicense Firefox, Thunderbird, and related software.

Author: I have a project which helps with identifying the existing copyright holders in your code, but I don't want to include it without prior discussion to avoid "conflicts of interest"/"self-promotion" concerns.

https://github.com/thehale/git-authorship

Reviewer 3: I don't think this would present a conflict of interest, if you still want to include it please feel free to do so!

* **Patents:** Your company may wish to join the [Open Invention Network](https://www.openinventionnetwork.com/), a shared defensive patent pool to protect members' use of major open source projects, or explore other [alternative patent licensing](https://www.eff.org/document/hacking-patent-system-2016).
* **Governance:** Especially if and when it makes sense to move a project to a [legal entity outside of the company](../leadership-and-governance/#do-i-need-a-legal-entity-to-support-my-project).

## What if someone uses my project without following the license terms?

<!-- 
  TODO 
   - Talk about **registering a copyright** to claim "statutory damages" since otherwise claims would only be able to cover "actual damages", which are much harder to argue for from a free, open-source project.
   - Talk about litigation costs. Consider mentioning foundations which help open source developers (FSF, EFF, sfconservancy)[https://github.com/community/maintainers/discussions/341#discussioncomment-6491105]

> -->

Author: The linked Maintainer Community discussion highlights what I consider this article's biggest gap -- how to handle infringement.

The obvious answer is "get a lawyer", but I think it would be helpful to provide links to OSS-specific foundations/communities that help with litigation.
